### PR TITLE
hex: enable interior facet integration

### PR DESCRIPTION
**Pull request by ksagiyam**
Add methods in `Cell` that return:

- a map from orientations to the corresponding axis permutation matrix.
- a map from orientations to the corresponding point/DoF permutation arrays.

### CODE REVIEW

**Comment by ksagiyam** [comment]
Restructured and moved some parts to FInAT.

File: FIAT/reference_element.py [hunk 1]
```diff
@@ -1131,6 +1227,70 @@ def __ge__(self, other):
     def __le__(self, other):
         return self.compare(operator.le, other)
 
+    def extract_extrinsic_orientation(self, o):
+        """Extract extrinsic orientation.
+
+        Parameters
+        ----------
+        o : object
```
Comment (dham) [inline FIAT/reference_element.py:1235]: I think we need a base class for `o` that tells us how `//` and `%` are supposed to behave.

Comment (ksagiyam) [reply FIAT/reference_element.py]: Added `Orientation` class.

**Comment by ksagiyam** [comment]
Reviewed in a Firedrake meeting.

**Pull request merged**